kolibri/plugins/learn/assets/src/state/actions/recommended.js

```diff
@@ -0,0 +1,272 @@
+  const resumePayload = { resume: currentUserId(state) };
+  const channelPayload = { channel_id: channelId };
+
+  // all you really need is user_id. Should we check in parent?
+  if (isFacilityUser(state)) {
+    return ContentNodeResource.getCollection(channelPayload, resumePayload).fetch();
+  }
+  return Promise.resolve([]);
+}
+
+function _getOverview(channelId, cursor) {
+  const channelPayload = { channel_id: channelId };
+
+  return ContentNodeResource.getAllContentCollection(channelPayload, { cursor }).fetch();
+}
+
+function showLearnChannel(store, channelId, cursor) {
+  // Special case for when only the page number changes:
+  // Don't set the 'page loading' boolean, to prevent flash and loss of keyboard focus.
+  const state = store.state;
+  if (state.pageName !== PageNames.LEARN_CHANNEL || getCurrentChannelId(state) !== channelId) {
+    store.dispatch('CORE_SET_PAGE_LOADING', true);
+  }
+
+  const channelsPromise = setChannelInfo(store, channelId);
+  channelsPromise.only(
+    samePageCheckGenerator(store),
+    () => {
+      if (!getCurrentChannelObject(state)) {
+        router.replace({ name: PageNames.CONTENT_UNAVAILABLE });
+        return;
+      }
+      ConditionalPromise.all([
+        _getNextSteps(channelId, state),
+        _getPopular(channelId),
+        _getResume(channelId, state),
+        _getOverview(channelId, cursor),
+      ]).only(
+        samePageCheckGenerator(store),
+        ([nextSteps, popular, resume, overview]) => {
+          const currentChannel = getCurrentChannelObject(store.state);
+          const pageState = {
+            // Hard to guarantee this uniqueness on the database side, so
+            // do a uniqBy content_id here, to prevent confusing repeated
+            // content items.
+            nextSteps: uniqBy(nextSteps, 'content_id').map(contentState),
+            popular: uniqBy(popular, 'content_id').map(contentState),
+            resume: uniqBy(resume, 'content_id').map(contentState),
+            overview: overview.map(contentState),
+            channelId: currentChannel.id,
+          };
+          store.dispatch('SET_PAGE_STATE', pageState);
+          store.dispatch('CORE_SET_PAGE_LOADING', false);
+          store.dispatch('CORE_SET_ERROR', null);
+          store.dispatch('SET_PAGE_NAME', PageNames.LEARN_CHANNEL);
+
+          store.dispatch('CORE_SET_TITLE', `Learn - ${currentChannel.title}`);
+        },
+        error => {
+          handleApiError(store, error);
+        }
+      );
+    },
+    error => {
+      handleApiError(store, error);
+    }
+  );
+}
+
+function showPopularPage(store, channelId) {
+  const state = store.state;
+
+  const pagePrep = Promise.all([
+    _getPopular(channelId, state),
+    setChannelInfo(store, channelId),
+  ]).then(([popular]) => popular, error => error);
+
+  pagePrep.then(
+    popularContent => {
+      const popularPageState = {
+        recommendations: uniqBy(popularContent, 'content_id').map(contentState),
```

Comment:
Alternatively, could just make a new helper function:
```
const mapContent = (contentArray) => {
    return uniqBy(contentArray, 'content_id').map(contentState);
}
```
Would save a fair amount of repeated code.